src/rust/engine/task_executor/src/lib.rs

```diff
@@ -78,10 +87,31 @@ impl Executor {
       });
     }
 
-    let runtime = Builder::new_multi_thread()
+    let mut runtime_builder = Builder::new_multi_thread();
+
+    runtime_builder
       .worker_threads(num_worker_threads)
       .max_blocking_threads(max_threads - num_worker_threads)
-      .enable_all()
+      .enable_all();
+
+    if env::var("PANTS_DEBUG").is_ok() {
```

Comment:
So, rather than adding a `pyo3` dep to the `task_executor` crate (which is used by other executables like `brfs`, `process_executor`, `fs_util`), it would be better to add an `on_thread_start: impl FnOnce() -> ()` argument to `fn global`, so that a caller can add this logic.